frontend/test/pytest/test_mid_circuit_measurement.py

```diff
@@ -570,6 +570,7 @@ def circuit(x):
         assert result.shape == (shots,)
         assert jnp.allclose(result, expected)
 
+    @pytest.mark.skip(reason="nested qnode not supported by catalyst runtime")
```

Comment:
This is depends on the target :sweat_smile:

Assuming:

* target is our runtime
* we are in full control
* we want to support sibling nodes with minimal changes

We can do sibling modules (as long as the qnodes are separable, i.e., the input to one qnode B is not a mid circuit measurement of qnode A and the output of qnode B is used by qnode A). Or we can just have a module with two qfunc's (which at the moment breaks the invariant, but I don't think it would be too difficult to deal with it). Or, we can do nested modules... That would also work, but I think semantically it would be something like the QPU can schedule execution on another QPU. This might be hard to model.

What do you think?

In my opinion, nested qnodes should be disallowed for the time being. The analogy is a GPU thread issuing another GPU to run and waiting for its output to continue executing. Is that possible?

Reply:
Just to keep the code generation aspect running, right now it will generate a third nested module. But I suggest we just disable calling a qnode from inside another qnode in the frontend and create an assertion.

Reply:
Closed with #1176